### PR TITLE
fix: ported to error catalog

### PR DESCRIPTION
**Pull request by gemaxim**
### What this does

Changed error to be from error-catalog so that they are properly handlled for dotnet-deps.

### CODE REVIEW

File: lib/lock-file-parser.ts [hunk 1]
```diff
@@ -7,11 +8,15 @@ export function packageSpecsFrom(
   try {
     lockFile = toml.parse(lockFileContents) as unknown as PoetryLockFile;
   } catch {
-    throw new LockFileNotValid();
+    throw new OpenSourceEcosystems.UnparseableLockFileError(
+      'The poetry.lock file is not parsable.',
+    );
```
Comment (dotkas) [inline lib/lock-file-parser.ts:13]: Can we add the error to the error args? This gives zero information in the logs.

File: lib/manifest-parser.ts [hunk 2]
```diff
@@ -12,7 +13,9 @@ export function pkgInfoFrom(manifestFileContents: string): PkgInfo {
       version: manifest.tool.poetry.version,
     };
   } catch {
-    throw new ManifestFileNotValid();
+    throw new OpenSourceEcosystems.UnparseableManifestError(
+      'The pyproject.toml file is not parsable.',
+    );
```
Comment (dotkas) [inline lib/manifest-parser.ts:18]: Same

File: lib/manifest-parser.ts [hunk 3]
```diff
@@ -26,11 +29,15 @@ export function getDependenciesFrom(
       manifestFileContents,
     ) as unknown as PoetryManifestType;
   } catch {
-    throw new ManifestFileNotValid();
+    throw new OpenSourceEcosystems.UnparseableManifestError(
+      'The pyproject.toml file is not parsable.',
+    );
```
Comment (dotkas) [inline lib/manifest-parser.ts:34]: Same

**Review by dotkas — CHANGES_REQUESTED** [review]
Just a small one

**Comment by gemaxim** [comment]
Added error to args @dotkas

**Review by dotkas — APPROVED** [review]

**Pull request merged**